app/js/services/situationService.js

```diff
@@ -226,6 +223,19 @@ angular.module('ddsApp').factory('SituationService', function($http, $sessionSto
                 .concat(enfants)
                 .concat(individus.slice(1));
             situation.individus = individus;
+        },
+
+        ressourcesYearMoins2Captured: function(situation) {
+            var ressourcesYM2Names = ['rncRevenusActivite', 'rncAutresRevenus', 'rncPensionsRetraitesRentes', 'fraisReelsDeductibles', 'rncPensionsAlimentaires', 'rncPensionsAlimentairesVersees'];
+            var ressourcesYM2 = [].concat.apply(
+                [],
+                situation.individus.map(function(individu) {
```

Comment:
To flatten the array of arrays returned by map. If I just return the array, the length will be the same as `individus` without checking whether they have N-2 ressources.
But I'm trying to find a more simple solution if this is too hard to grasp.